gtk/main.cc

```diff
@@ -50,6 +50,9 @@ Glib::OptionEntry create_option_entry(Glib::ustring const& long_name, gchar shor
 
 int main(int argc, char** argv)
 {
+    /* manage cURL global constants */
+    auto const curl_mgr = tr_get_curl_mgr();
```

Comment:
Could this be abstracted away as some sort of "libtransmission initialization"? You are spilling out internal implementation details with this sort of naming.

Reply:
It could also absorb `tr_net_init()`, and then that function name could be removed from the headers

Reply:
Something like `tr_init()` or `libtransmission::init()`, yes?